src/Aspire.Hosting/MongoDB/IMongoDbResource.cs

```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents a MongoDb resource that requires a connection string.
+/// </summary>
+public interface IMongoDbResource : IResourceWithConnectionString
```

Comment:
I suggest we keep using `MongoDB` with uppercase letters in both code and comments since this is already used in package names, drivers, and this is how the official name is written.

Reply:
@eerhardt @DamianEdwards Has there been any discussion already about using `MongoDB` or `MongoDb` in class names? I believe the dotnet convention is `Db` but here it's the name of the product which has `DB`.

Reply:
With RabbitMQ we used the casing of the underlying client library to ensure it felt coherent.

https://github.com/dotnet/aspire/blob/df636ce6f10c09b56781c28204b37d96ed7e0dfc/src/Components/Aspire.RabbitMQ.Client/AspireRabbitMQExtensions.cs#L37

https://github.com/dotnet/aspire/blob/df636ce6f10c09b56781c28204b37d96ed7e0dfc/src/Aspire.Hosting/RabbitMQ/RabbitMQContainerResource.cs#L11

Reply:
The have few classes using `MongoDB` in their name, example `MongoDBMessage`. The namespace is `MongoDB`. It has some classes with `Mongo` only, e.g., `MongoGridFS`

But nothing with the case `Db.

Reply:
It looks like the framework has slightly higher prevalence of `Db`; but I'd agree that our general principle is "when in Rome...", so if the users would be interacting with things cased as `MongoDB` then I'd say use that casing.  (SHA, RSA, and HMAC all should have been Sha, Rsa, Hmac; but since they're already there we keep new types that interact with them using consistent casing)

Reply:
So, the idea is to follow the pattern with case and change all the references to `MongoDB`, right?

Reply:
I've changed all the references to use the casing ("MongoDB").